controllers/lvmcluster_controller.go

```diff
@@ -208,16 +208,33 @@ func (r *LVMClusterReconciler) reconcile(ctx context.Context, instance *lvmv1alp
 		&topolvmVolumeSnapshotClass{},
 	}
 
-	// handle create/update
-	for _, unit := range resourceCreationList {
-		err := unit.ensureCreated(r, ctx, instance)
-		if err != nil {
-			r.Log.Error(err, "failed to reconcile", "resource", unit.getName())
-			return ctrl.Result{}, err
+	resourceSyncStart := time.Now()
+	results := make(chan error, len(resources))
+	create := func(i int) {
+		if err := resources[i].ensureCreated(r, ctx, instance); err != nil {
+			results <- err
+		}
+	}
+
+	for i := range resources {
+		go create(i)
+	}
+
+	var errs []error
+	for i := 0; i < len(resources); i++ {
+		if err := <-results; err != nil {
```

Comment:
If you look at the code path there should not be a way to get stuck here. This will always return exactly the amount of times that a resource is attempted to be created. This means that there will always be the correct amount of results, no matter the amount of failures in the goroutines.